cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/internal/AliasConfiguration.qll

```diff
@@ -95,3 +98,31 @@ class IndirectParameterAllocation extends Allocation, TIndirectParameterAllocati
 
   final override predicate alwaysEscapes() { none() }
 }
+
+class DynamicAllocation extends Allocation, TDynamicAllocation {
+  CallInstruction call;
+
+  DynamicAllocation() { this = TDynamicAllocation(call) }
+
+  final override string toString() {
+    result = call.toString() + " at " + call.getLocation() // TODO: make this both short and unique
+  }
+
+  final override CallInstruction getABaseInstruction() { result = call }
+
+  final override IRFunction getEnclosingIRFunction() { result = call.getEnclosingIRFunction() }
+
+  final override Language::Location getLocation() { result = call.getLocation() }
+
+  final override string getUniqueId() { result = call.getUniqueId() }
+
+  final override IRType getIRType() {
+    result instanceof IRUnknownType // TODO: look at casts and sizes?
```

Comment:
This didn't seem to cause any issues in the tests, even with `DefaultTaintTracking` enabled. Would there be any benefit from having a precise type here?

Reply:
There's no more precise type that we could assign here, unless we knew that the allocation size was a constant. The constant case would show up more often that you may expect, because `new Foo` includes `void* p = operator new(sizeof(Foo));` in its expansion, Even then, though, I'd expect little if any benefit from having a more precise type there.